Simbody/src/GeneralForceSubsystem.cpp

```diff
@@ -211,52 +472,88 @@ class GeneralForceSubsystemRep : public ForceSubsystem::Guts {
         // force element is enabled.
         const Array_<bool>& forceEnabled = Value< Array_<bool> >::downcast
                                     (getDiscreteVariable(s, forceEnabledIndex));
+        const Array_<bool>& parallelEnabled = Value< Array_<bool> >::downcast
+                                    (getDiscreteVariable(s, parallelEnabledIndex));
 
         // Get access to System-global force cache arrays.
-        Vector_<SpatialVec>&   rigidBodyForces = 
+        Vector_<SpatialVec>&   rigidBodyForces =
                                     mbs.updRigidBodyForces(s, Stage::Dynamics);
-        Vector_<Vec3>&         particleForces  = 
+        Vector_<Vec3>&         particleForces  =
                                     mbs.updParticleForces (s, Stage::Dynamics);
-        Vector&                mobilityForces  = 
+        Vector&                mobilityForces  =
                                     mbs.updMobilityForces (s, Stage::Dynamics);
 
+        // Track the enabled forces and whether they should be parallelized.
```

Comment:
I don't think this should be done at Dynamics stage where it is called very frequently. It should only be done if `forceEnabled` changes (Instance stage). I don't understand why `parallelEnabled` is a state variable, but even if it is changing it will invalidate Instance stage so it can be fully processed during realizeInstance().